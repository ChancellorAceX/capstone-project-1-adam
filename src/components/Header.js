import React from 'react';
import { withRouter } from 'react-router';
import '../styles/Header.css';
import backButton from '../resources/back-2.png';
// import addButton from '../resources/add-2.png';
import logoutButton from '../resources/logout-3.png';
import TokenService from '../services/token-service';

export default withRouter(class Header extends React.Component {
  constructor(props) {
    super(props);
    this.goBack = this.goBack.bind(this);
  }
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  };

  goBack() {
    this.props.history.goBack();
  }
  logoutHandler=(e)=> {
    TokenService.clearAuthToken()
    this.props.loggedInToggle(false)
    const { history } = this.props;
    const destination = '/login';
    history.push(destination);
  }

  render() {
    return (
      <header className='bdheader'>
        <img className='backButton' onClick={this.goBack} src={backButton} alt='Back Button' />
        <h1>{this.props.pageName}</h1>
        <img className='logoutButton' src={logoutButton} alt='Logout Button' onClick={this.logoutHandler} />
      </header>
    );
  }
})