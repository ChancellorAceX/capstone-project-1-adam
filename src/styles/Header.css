header {
  position:fixed;
  top:0;
  height:15vh;
  border-bottom:2px solid black;
  background-color: #333333;
}

header h1 {
  text-align: center;
  margin:5px;
  color:#ba0404;
  text-shadow: 2px 2px 5px black;
}

header img {
  cursor:pointer;
}