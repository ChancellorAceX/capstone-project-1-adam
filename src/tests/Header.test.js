import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../components/Header';

describe('Header component', () => {
  const props = {
    "pageName": "Encounters",
    "history": {
      "length": 8,
      "action": "PUSH",
      "location": "{hash: \"\", key: \"bhl6me\", pathname: \"/encounters\", …}",
      "createHref": "ƒ createHref() {}",
      "push": "ƒ push() {}",
      "replace": "ƒ replace() {}",
      "go": "ƒ go() {}",
      "goBack": "ƒ goBack() {}",
      "goForward": "ƒ goForward() {}",
      "block": "ƒ block() {}",
      "listen": "ƒ listen() {}"
    },
    "loggedInToggle": "ƒ () {}"
  };

  it('renders the full header component', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});