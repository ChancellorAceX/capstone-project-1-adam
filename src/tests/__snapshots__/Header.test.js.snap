// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the full header component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
